core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -878,104 +926,119 @@ func applyAdditionalQueryOptions(queryString string, queryLimit int32, queryBook
 }
 
 type queryScanner struct {
-	namespace       string
-	db              *couchDatabase
-	queryDefinition *queryDefinition
-	paginationInfo  *paginationInfo
-	resultsInfo     *resultsInfo
-	exhausted       bool
+	namespace   string
+	db          *couchDatabase
+	queryDef    *queryDefinition
+	pageInfo    *paginationInfo
+	resultsInfo *resultsInfo
+	exhausted   bool
 }
 
 type queryDefinition struct {
-	startKey           string
-	endKey             string
-	query              string
-	internalQueryLimit int32
+	startKey string
+	endKey   string
+	query    string
+	limit    int32
```

Comment:
Ideally, this struct is a wrong place for this `internalQueryLimit` as this is more of a config param as opposed to anything to do with `queryDefinition`. Keeping that in mind, this name change further dilutes that meaning and gives an impression as if this is the limit of the query results. Perhaps, it would be better to remove this field altogether from this struct.